### PR TITLE
use pasteHTML & refactor a bit for URL use case

### PR DESCRIPTION
**Pull request by martinklepsch**
@belucid this is just the commit I said you could cherry pick, deployed this to staging for Stuart

### CODE REVIEW

**Comment by bago2k4** [comment]
reviewing

File: src/open_company_web/components/topic_overlay_edit.cljs [hunk 1]
```diff
@@ -231,49 +231,56 @@
                           with-finances-data)]
     with-growth-data))
 
-(defcomponent uploader [data owner]
+(defn upload-file! [editor owner file]
+  (let [success-cb  (fn [success]
+                      (js/console.log success)
+                      (js/console.log editor)
```
Comment (bago2k4) [inline src/open_company_web/components/topic_overlay_edit.cljs]: @martinklepsch do we need these logs?

**Comment by bago2k4** [comment]
@martinklepsch can you add test steps?
since stuart needs this i'll create a new branch where i'll merge my work on the new editing component and this and deploy it to staging.
We can discuss this tomorrow or monday

File: src/open_company_web/components/topic_overlay_edit.cljs [hunk 2]
```diff
@@ -231,49 +231,54 @@
...
-                                                            (fn [success]
-                                                              (js/MediumEditor.util.insertHTMLCommand js/document (str "<img src=\"" (.-url success) "\">"))
-                                                              (js/console.log (om/get-node owner))
-                                                              ;; (gstyle/setElementShown (om/get-node owner) false)
-                                                              (om/set-state! owner {}))
-                                                            (fn [error] (js/console.log "error" error))
-                                                            #(om/set-state! owner {:state :show-progress
-                                                                                   :progress %})))})
+                    :on-change #(upload-file! editor owner (-> % .-target .-files (aget 0)))})
         (dom/button {:style {:margin-right "13px"}
```
Comment (belucid) [inline src/open_company_web/components/topic_overlay_edit.cljs]: @martinklepsch why do we have all this inline styling?

File: src/open_company_web/components/topic_overlay_edit.cljs [hunk 3]
```diff
@@ -231,49 +231,54 @@
...
                      :on-click (fn [_] (om/set-state! owner :state :show-options))}
-                    (i/icon :circle-add {:size 24}))
+          (i/icon :circle-add {:size 24}))
         (case (:state (om/get-state owner))
           :show-options
           (dom/div (dom/button {:style {:font-size "14px"} :class "underline"
```
Comment (belucid) [inline src/open_company_web/components/topic_overlay_edit.cljs]: and here

File: src/open_company_web/components/topic_overlay_edit.cljs [hunk 4]
```diff
@@ -231,49 +231,54 @@
...
                                 :on-click #(.click (js/document.getElementById "file-upload-ui--select-trigger"))}
-                               "Select an image")
-                   #_(dom/span {:style {:font-size "14px"}} " or ")
-                   #_(dom/button {:style {:font-size "14px"} :class "underline"
-                                :on-click #(om/set-state! owner :state :show-url-field)}
-                               "provide URL"))
+                     "Select an image")
+            #_(dom/span {:style {:font-size "14px"}} " or ")
```
Comment (belucid) [inline src/open_company_web/components/topic_overlay_edit.cljs]: and here

File: src/open_company_web/components/topic_overlay_edit.cljs [hunk 5]
```diff
@@ -231,49 +231,54 @@
...
           :show-url-field
           (dom/div (dom/input {:type "text" :auto-focus true :on-change #(do (om/set-state! owner :url (-> % .-target .-value)) true)
                                :value (om/get-state owner :url)})
-                   (dom/button {:style {:font-size "14px" :margin-left "1rem"} :class "underline"
-                                :on-click #(do (js/MediumEditor.util.insertHTMLCommand js/document (str "<img src=\"" (om/get-state owner :url) "\">"))
-                                               (om/set-state! owner {}))}
-                               "add")
-                   )
-          (dom/div "")))))
+            (dom/button {:style {:font-size "14px" :margin-left "1rem"} :class "underline"
```
Comment (belucid) [inline src/open_company_web/components/topic_overlay_edit.cljs]: and here

**Comment by martinklepsch** [comment]
Sorry about the lack of test steps, initially opened this for Sean who knew what it was about. Basically 

- on staging or production
- edit a topic 
- upload an image 
- check that the image is shown

**Comment by belucid** [comment]
@martinklepsch If you click the + but then don’t go through with it and U/L an image, then you don’t see the + on each new line anymore, you see the Select an image link and it gets in the way of your ability to write.

It stays, essentially (+) selected until you cancel out of the editing mode, or you actually do U/L an image

**Comment by martinklepsch** [comment]
Regarding inline styling: I initially wanted to use classes like `h5` & `mr1` but the CSS for that component has rules like this:
```
div.topic-overlay-edit button {
    background-color: transparent;
    border: 0px;
    margin: 0;
    padding: 0;
    outline: none;
    font-size: 12px;
}
```
which you can only override  if you add more selectors with higher specificity. Because I think that's not a maintainable solution (see [this](http://mrmrs.io/writing/2016/03/24/scalable-css/)) I just used inline styles.

In general I think inline styles in JS apps are much less bad then they are in plain html.

**Comment by martinklepsch** [comment]
@belucid the expanded state can now be toggled by clicking the plus again. I also added a small animation to make that a bit easier to spot. Automatically collapsing on cursor movement is unfortuntately a bit tricky so I hope this is sufficient.

**Comment by belucid** [comment]
@martinklepsch and @bago2k4 This is in `staging-test` branch which will ultimately merge into `the-revolution-dashboard` and be PR'd there, so closing this.

**Pull request closed without merging**